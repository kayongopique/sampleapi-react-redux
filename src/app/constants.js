export const REQUEST_WORK_ORDERS = 'REQUEST_WORK_ORDERS';
export const TOGGLE_SWITCH = 'TOGGLE_SWITCH';
export const RECIEVE_WORK_ORDERS = 'RECIEVE_WORK_ORDERS';
export const SEARCH_INPUT = 'SEARCH_INPUT'