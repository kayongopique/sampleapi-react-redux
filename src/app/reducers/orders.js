import { RECIEVE_WORK_ORDERS, SEARCH_INPUT, TOGGLE_SWITCH} from '../constants'; 
import findOrders from '../utilities/filter';
import sortAscending from '../utilities/sortAsc';
import sortDescending from '../utilities/sortDesc'


const orders = (state = [], action) => {
    switch (action.type) { 
        case RECIEVE_WORK_ORDERS:
            return {
                ...state,
                orders: action.payload.orders? 
                sortAscending(action.payload.orders ): sortAscending(action.payload.orders ),
                errors: action.payload.errors ? action.payload.errors : null, 
            }
    
        case SEARCH_INPUT:
           let  payload = state.orders;
            state = {
                searchItem: '',
                orders: payload
            }
            const searchItem = action.searchItem;
            const orders = state.orders;
        return {
            ...state,
            searchItem,
            filtered_orders: searchItem ? findOrders(orders, searchItem) : orders
            
        }
        case TOGGLE_SWITCH:
                const orders_arr = state.orders;
                state = {
                    tab: false,
                    orders: orders_arr
                }
            const tab = action.tab;
        return {
            ...state,
            tab,
            orders: tab ? sortDescending(state.orders) : sortAscending(state.orders)    
        }
        default:       
            return state;   
        } 
    };  
    
export default orders; 