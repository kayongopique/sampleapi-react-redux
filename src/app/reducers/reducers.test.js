import orders from './orders'; 
import { RECIEVE_WORK_ORDERS, SEARCH_INPUT, TOGGLE_SWITCH} from '../constants'; 
import  deepFreeze  from 'deep-freeze';

describe(' Orders Reducer ', () => {
    const initialstate = deepFreeze(orders(undefined, {}));

    it(' should return default state on unknown actions  ', () => {
        expect(orders(initialstate, {type: 'UNKNOWN'})).toEqual(initialstate);
    });

    it('Should add orders to empty initialState ', () => {
        expect(orders(initialstate, {type: RECIEVE_WORK_ORDERS, 
                payload:{"orders": [{
                                "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                               "deadline": "2012-09-06 09:02:23"
                            } ,
                            {
                                "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                               "deadline": "2012-09-06 09:02:23"
                            },
                            {
                                "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                               "deadline": "2012-09-06 09:02:23"
                            } ]}
                         })).toMatchSnapshot();
    });

    it('Should add orders to non empty initialState ', () => {
        const baseState = deepFreeze(orders(initialstate, {type: RECIEVE_WORK_ORDERS,
                    payload: {"orders": [{
                        "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                       "deadline": "2012-09-06 09:02:23"
                    } ,
                    {
                        "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                       "deadline": "2012-09-06 09:02:23"
                    },
                    {
                        "worker": {"worker":{"name":"kayongo david","company": "cisco"}},
                       "deadline": "2012-09-06 09:02:23"
                    } ]}}));
        expect(orders(baseState, {type: RECIEVE_WORK_ORDERS, 
                payload: {"orders": [{
                    "worker": {"worker":{"name":"paul david","company": "cisco"}},
                   "deadline": "2012-09-06 09:02:23"
                } 
                 ]} })).toMatchSnapshot();
    });

});

