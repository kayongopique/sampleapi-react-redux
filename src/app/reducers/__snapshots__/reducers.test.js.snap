// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Orders Reducer  Should add orders to empty initialState  1`] = `
Object {
  "errors": null,
  "orders": Array [
    Object {
      "deadline": "2012-09-06 09:02:23",
      "worker": Object {
        "worker": Object {
          "company": "cisco",
          "name": "kayongo david",
        },
      },
    },
    Object {
      "deadline": "2012-09-06 09:02:23",
      "worker": Object {
        "worker": Object {
          "company": "cisco",
          "name": "kayongo david",
        },
      },
    },
    Object {
      "deadline": "2012-09-06 09:02:23",
      "worker": Object {
        "worker": Object {
          "company": "cisco",
          "name": "kayongo david",
        },
      },
    },
  ],
}
`;

exports[` Orders Reducer  Should add orders to non empty initialState  1`] = `
Object {
  "errors": null,
  "orders": Array [
    Object {
      "deadline": "2012-09-06 09:02:23",
      "worker": Object {
        "worker": Object {
          "company": "cisco",
          "name": "paul david",
        },
      },
    },
  ],
}
`;
